### PR TITLE
[public-api] Create token UI

### PR DESCRIPTION
**Pull request by jeanp413**
## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/14615

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/dashboard/src/settings/PersonalAccessTokens.tsx [hunk 1]
```diff
@@ -27,31 +35,218 @@ function PersonalAccessTokens() {
...
+        if (value.name.length === 0) {
+            setErrorMsg("Token Name must not be empty.");
+            return;
+        }
+        try {
+            const resp = await personalAccessTokensService.createPersonalAccessToken({
+                token: {
+                    name: value.name,
+                    expirationTime: Timestamp.fromDate(value.expirationDate),
+                    scopes: ["*"],
```
Comment (easyCZ) [inline components/dashboard/src/settings/PersonalAccessTokens.tsx:80]: With https://github.com/gitpod-io/gitpod/pull/14889, this should either be empty, or contain exactly this array `["function:*", "resource:default"]`

**Comment by easyCZ** [comment]
This is awesome!

Attaching screenshots just for completness:
Screenshot 2022-11-24 at 12 58 47
Screenshot 2022-11-24 at 12 58 28
Screenshot 2022-11-24 at 12 58 23

File: components/dashboard/src/settings/PersonalAccessTokens.tsx (hunk 1 above)
Comment (easyCZ) [reply components/dashboard/src/settings/PersonalAccessTokens.tsx:80]:
```suggestion
                    scopes: ["function:*", "resource:default"],
```

**Review by easyCZ — APPROVED** [review]
LGTM.

I'm not focusing on the UI/UX polish too much at this point as making it functional first is the priority. We can do another pass on UX adjustments later.

Feel free to remove the hold once the scopes are updated

**Comment by gtsiolis** [comment]
> I'm not focusing on the UI/UX polish too much at this point as making it functional first is the priority. We can do another pass on UX adjustments later.

ok

**Comment by easyCZ** [comment]
I've applied the suggested fix so we can land this. It remains behind a feature flag so future iterations to fix up are easy to do. Will unhold once the build passes.

**Comment by mustard-mh** [comment]
Rebased to main, going to do some 🧹 

/hold

**Review by mustard-mh — APPROVED** [review]
LGTM

/unhold

**Pull request merged**